dace/frontend/python/newast.py

```diff
@@ -1236,6 +1236,11 @@ def _views_to_data(state: SDFGState, nodes: List[dace.nodes.AccessNode]) -> List
                     if (pyname and dtypes.validate_name(pyname) and pyname not in self.sdfg.arrays):
                         self.sdfg.replace(arrname, pyname)
 
+        propagate_states(self.sdfg)
```

Comment:
This is probably fine but I wonder whether we are doing duplicate operations. For example, in the `_add_dependencies` method, the symbols are iteratively propagated anyway and perhaps we could set there the `memlet.dynamic` property. The cost might be negligible anyway, but it would be good to note this down for future optimizations, in case it is not.